src/middlewared/middlewared/plugins/pool_/dataset.py

```diff
@@ -657,6 +662,24 @@ async def do_create(self, app, data):
             )
             await acl_job.wait()
 
+        elif data['type'] == 'FILESYSTEM' and data['share_type'] == 'APPS':
+            acl = (await self.middleware.call('filesystem.acltemplate.by_path', {
+                'query-filters': [('name', '=', 'NFS4_RESTRICTED')],
+                'format-options': {'canonicalize': True, 'ensure_builtins': True},
+            }))[0]['acl']
+            acl.append({
+                'tag': 'USER',
+                'id': 568,
```

Comment:
Just a nit, but maybe define this UID somewhere (maybe we already do and I'm just forgetting)